### PR TITLE
feat: email protection saving

### PR DESCRIPTION
**Pull request by shakyShane**
**Reviewer:** @GioSensation 
**Asana:** https://app.asana.com/0/0/1204988996468240/f

## Description

- [x] on macOS we now send an additional `storeFormData` message _only_ when you select a private address
- [x] we remember what the last email address that we generated was - if that matches in the form submission, we set the `autogenerated` flag

## Breaking change:

- [ ] before windows can upgrade, we need to check if this additional message will cause any issues - I'll document this in Asana and help them to test

## Steps to test

- This has been tested extensively on macOS + Android during the implementation

### CODE REVIEW

**Comment by shakyShane** [comment]
Current dependencies on/for this PR:
* main
  * **PR #355** <a href="https://app.graphite.dev/github/pr/duckduckgo/duckduckgo-autofill/355" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/duckduckgo-autofill/355?utm_source=stack-comment).

File: src/EmailProtection.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    /** @type {string|null} */
+    #previous = null;
+
+    /** @param {import("./DeviceInterface/InterfacePrototype").default} device */
+    constructor (device) {
+        this.device = device
+    }
+
+    /** @returns {boolean} */
+    get generated () {
```
Comment (GioSensation) [inline src/EmailProtection.js:14]: Should this be `isGenerated` or `hasGenerated`? At first glance, I was expecting it to hold the actual value.

File: src/EmailProtection.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    /** @returns {string|null} */
+    get lastGenerated () {
+        return this.#previous
+    }
+
+    /**
+     * Store the last received email address
+     * @param {string} emailAddress
+     */
+    received (emailAddress) {
```
Comment (GioSensation) [inline src/EmailProtection.js:27]: This name, too, could be confusing. How about `storeReceived`?

**Review by GioSensation — APPROVED** [review]
A couple of minor naming tweaks, but otherwise this looks grand. Thank you! 🚀

**Pull request merged**